geth/account/accounts_test.go

```diff
@@ -109,3 +111,46 @@ func TestVerifyAccountPasswordWithAccountBeforeEIP55(t *testing.T) {
 	_, err = acctManager.VerifyAccountPassword(keyStoreDir, address.Hex(), TestConfig.Account3.Password)
 	require.NoError(t, err)
 }
+
+func TestCreateAndRecoverAccount(t *testing.T) {
+	ctrl := gomock.NewController(t)
+	nodeManager := common.NewMockNodeManager(ctrl)
+
+	// Create an arbitrary keystore for the process
+	keyStoreDir, err := ioutil.TempDir(os.TempDir(), "accounts")
+	require.NoError(t, err)
+	defer os.RemoveAll(keyStoreDir) //nolint: errcheck
+	keyStore := keystore.NewKeyStore(keyStoreDir, 4, 5)
+
+	manager := account.NewManager(nodeManager)
+	password := "some-pass"
```

Comment:
I would also verify empty string as a password. I think it may be ok but let's make sure as it's a special case. You can also think about extracting various password cases to separate `TestCreateAccount()` function.

Reply:
I actually did this one but removed later :)
Using an empty string doesn't make a difference in the result, it throws no error and returns a valid result. Shall we also expect it to not fail with an empty password in tests? Or slightly extend the scope of this PR and modify the function to fail?

Reply:
> Shall we also expect it to not fail with an empty password in tests?

No, this is good I think, at least for now. My point is to just make a unit test for this scenario so that if it changes, we're aware as this is an edge case.